logfire-api/logfire_api/__init__.py

```diff
@@ -0,0 +1,167 @@
+from __future__ import annotations
+
+from contextlib import contextmanager
+import importlib
+import sys
+from typing import TYPE_CHECKING, Literal
+from unittest.mock import MagicMock
+
+try:
+    logfire_module = importlib.import_module('logfire')
+    sys.modules[__name__] = logfire_module
+
+except ImportError:
+    if not TYPE_CHECKING:  # pragma: no branch
+        LevelName = Literal['trace', 'debug', 'info', 'notice', 'warn', 'warning', 'error', 'fatal']
+        VERSION = '0.0.0'
+        METRICS_PREFERRED_TEMPORALITY = {}
+
+        def configure(*args, **kwargs): ...
+
+        class LogfireSpan:
+            def __getattr__(self, attr):
+                return MagicMock()  # pragma: no cover
+
+            def __enter__(self):
+                return self
+
+            def __exit__(self, *args, **kwargs) -> None: ...
+
+        class Logfire:
+            def __getattr__(self, attr):
+                return MagicMock()  # pragma: no cover
+
+            def __init__(self, *args, **kwargs) -> None: ...
+
+            def span(self, *args, **kwargs) -> LogfireSpan:
+                return LogfireSpan()
+
+            def log(self, *args, **kwargs) -> None: ...
+
+            def trace(self, *args, **kwargs) -> None: ...
+
+            def debug(self, *args, **kwargs) -> None: ...
+
+            def notice(self, *args, **kwargs) -> None: ...
+
+            def info(self, *args, **kwargs) -> None: ...
+
+            def warn(self, *args, **kwargs) -> None: ...
+
+            def error(self, *args, **kwargs) -> None: ...
+
+            def fatal(self, *args, **kwargs) -> None: ...
+
+            def with_tags(self, *args, **kwargs) -> Logfire:
+                return self
+
+            def with_settings(self, *args, **kwargs) -> Logfire:
+                return self
+
+            def force_flush(self, *args, **kwargs) -> None: ...
+
+            def log_slow_async_callbacks(self, *args, **kwargs) -> None: ...  # pragma: no branch
+
+            def install_auto_tracing(self, *args, **kwargs) -> None: ...
+
+            def instrument(self, *args, **kwargs):
+                def decorator(func):
+                    return func
+
+                return decorator
+
+            def instrument_fastapi(self, *args, **kwargs) -> None: ...
```

Comment:
This, instrument_openai, and maybe others should be a context manager. Might as well make the all return nullcontext.